internal/authentication/user_provider.go

```diff
@@ -1,9 +1,14 @@
 package authentication
 
+import (
+	"time"
+)
+
 // UserProvider is the interface for checking user password and
 // gathering user details.
 type UserProvider interface {
 	CheckUserPassword(username string, password string) (bool, error)
 	GetDetails(username string) (*UserDetails, error)
 	UpdatePassword(username string, newPassword string) error
+	GetRefreshSettings() (enabled bool, interval time.Duration)
```

Comment:
So I think there are a few ways to handle this. Looking for some guidance though. I was thinking instead of including the verifySessionIsUpToDate() func in handler_verify.go we make it part of the either the session or auth provider. Which ever one would also hold the configuration settings and we could do away with that GetRefreshSettings().